core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -38,6 +38,28 @@ static size_t get_ecc_key_size_bytes(uint32_t curve)
 	}
 }
 
+/*
+ * Returns the key size in bits for the given ECC curve
+ *
+ * @curve   ECC Curve ID
+ */
+static size_t get_ecc_key_size_bits(uint32_t curve)
+{
+	switch (curve) {
+	case TEE_ECC_CURVE_NIST_P192:
+	case TEE_ECC_CURVE_NIST_P224:
+	case TEE_ECC_CURVE_NIST_P256:
+	case TEE_ECC_CURVE_NIST_P384:
+		return 8 * get_ecc_key_size_bytes(curve);
+
+	case TEE_ECC_CURVE_NIST_P521:
+		return 521;
```

Comment:
For consistency of sizes defined in `get_ecc_key_size_bits()` and `get_ecc_key_size_bytes()`, maybe define sizes only in `get_ecc_key_size_bits()` and have `get_ecc_key_size_bytes()` do the conversion?

Reply:
yes I know but I didnt like the way it was done before (and which I have replicated in other places)...IMO this is much more clear and easier to read as it immediately flags a curve that needs to be handled with care - some HW ciphers might require padding or tricks.
unless it is a strong request  from the file maintainer I'd rather keep it this way.

Reply:
I agree it certainly flags that something funny is going on, but I'd prefer if that was explained in a good comment since it's not obvious what it is from this.